Core/src/main/java/com/github/ezauton/core/action/ActionGroup.java

```diff
@@ -182,10 +183,12 @@ public Action getAction() {
     @Override
     public final void run(ActionRunInfo actionRunInfo) throws ExecutionException {
         List<WithActionData> withActions = new ArrayList<>();
-//        List<Future<Void>> withActionFutures = new ArrayList<>();
         List<Future<Void>> actionFutures = new ArrayList<>();
 
         for (ActionWrapper scheduledAction : scheduledActions) {
+            if(interrupted) { // stop scheduling new actions
+                break;
```

Comment:
This does fix it... however, it will still not cancel the blocked actions right away which imo bad implementation. I think instead of relying on an `interrupted()` method, all `Action`s should just rely on `InterruptedException` like normal Java threads. Actions which normally loop (`PeriodicAction`) could have an additional `interrupted()` method, but this would be part of the `PA` impl, not part of the `Action` interface.